spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.util
+
+import scala.util.Random
+
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.delta.actions.CommitInfo
+import org.apache.spark.sql.delta.stats.DataSize
+import com.fasterxml.jackson.core.StreamReadConstraints
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.test.SharedSparkSession
+
+class JsonUtilsSuite
+  extends QueryTest
+  with SharedSparkSession {
+
+  test("DataSize json serialization") {
+    val testCases = Seq(
+      DataSize() -> """{}""",
+      DataSize(bytesCompressed = Some(816L)) -> """{"bytesCompressed":816}""",
```

Comment:
Can you add a bit more variance in the data by adding negative long and 0L?

Reply:
the negative long is maybe more appropriate for the other test, since this one is data size, which shouldn't be negative.

Reply:
I added 0 and negative. While neagtive sizes are not valid, serialization should still handle them